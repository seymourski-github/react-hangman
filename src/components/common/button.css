.button {
  cursor: pointer;
  height: 2em;
}

.button[disabled],
.input[disabled] {
  cursor: no-drop;
  opacity: 0.6;
}
