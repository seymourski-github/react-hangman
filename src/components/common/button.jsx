import React from 'react';
import PropTypes from 'prop-types';
import { compose, defaultProps, setPropTypes, withHandlers } from 'recompose';
import { withStyleProps, withFlexItemProps } from './enhancers';
import './button.css';

const BaseButton = props => {
  const {
    className, disabled, hidden, id, name,
    onClick, style, text, type, value
  } = props;
  return (
    <button className={className} name={name}
      id={id||`${name}-button`}
      children={text||name}
      disabled={disabled}
      hidden={hidden}
      onClick={onClick}
      style={style}
      type={type}
      value={value}
    />
  );
};

export const Button = compose(
  defaultProps({
    className: 'button',
    disabled: false,
    type: 'button',
    handleClick: (e) => false
  }),
  setPropTypes({
    action: PropTypes.string,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    handleClick: PropTypes.func,
    id: PropTypes.string,
    name: PropTypes.string.isRequired,
    text: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string,
  }),
  withHandlers({
    onClick: props => e => {
      const { action, name, value, handleClick } = props;
      return action ? handleClick({action, name, value}) : handleClick(e);
    }
  }),
  withStyleProps,
  withFlexItemProps
)(BaseButton);
