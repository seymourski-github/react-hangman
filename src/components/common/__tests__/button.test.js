import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Button } from '../button';

const props = {
  name: "myButton"
};

describe('<Button /> Component', function() {

  const component = renderer.create(<Button {...props} />);
  let wrapper = shallow(<Button {...props} />);

  it("renders and matches our snapshot", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
