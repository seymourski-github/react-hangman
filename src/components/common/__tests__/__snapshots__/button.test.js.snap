// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Component renders and matches our snapshot 1`] = `
<button
  className="button"
  disabled={false}
  id="myButton"
  name="myButton"
  onClick={[Function]}
  style={null}
  type="button"
  value=""
>
  myButton
</button>
`;
